geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseQueryPlan.scala

```diff
@@ -107,34 +113,14 @@ object HBaseQueryPlan {
 
     override type Results = Result
 
-    override def scan(ds: HBaseDataStore): CloseableIterator[Result] = {
-      // note: we have to copy the ranges for each scan, except the last one (since it won't conflict at that point)
-      val iter = tables.iterator
-      val scans = iter.map(singleTableScan(ds, _, copyScans = iter.hasNext))
-
-      if (PartitionParallelScan.toBoolean.contains(true)) {
-        // kick off all the scans at once
-        scans.foldLeft(CloseableIterator.empty[Result])(_ ++ _)
-      } else {
-        // kick off the scans sequentially as they finish
-        SelfClosingIterator(scans).flatMap(s => s)
-      }
-    }
-
-    private def singleTableScan(
-        ds: HBaseDataStore,
-        table: TableName,
-        copyScans: Boolean): CloseableIterator[Result] = {
-      val s = if (copyScans) { scans.map(new Scan(_)) } else { scans }
-      HBaseBatchScan(ds.connection, table, s, ds.config.queryThreads)
-    }
+    override protected def singleTableScan(ds: HBaseDataStore, scan: TableScan): CloseableIterator[Result] =
+      HBaseBatchScan(ds.connection, scan.table, scan.scans, ds.config.queryThreads)
```

Comment:
Dumb question time:  It looks like we could almost return HBaseBatchScans and CoprocessorBatchScans instead of ScanPlans and CoprocessorPlans. 

I imagine that it is not the case.  Why not:)?

Reply:
we separate out creating the query plan from actually running it. that lets us do things like  configure spark input formats and explain queries without running them